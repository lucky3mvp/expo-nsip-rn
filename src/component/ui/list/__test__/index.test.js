import React from 'react'
import { Text } from 'react-native'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { List, ListItem } from '../index'

describe('List Tests', () => {
	describe('List test', () => {
		it('renders correctly', () => {
			const tree = Renderer.create((
				<List>
					<ListItem title='ListItem' />
				</List>
			)).toJSON();
			
			expect(tree).toMatchSnapshot();
		});
	});

	describe('ListItem test', () => {
		it('renders correctly', () => {
			const tree = Renderer.create((
				<ListItem title='ListItem' />
			)).toJSON();
			
			expect(tree).toMatchSnapshot();
		});

		it('press event', () => {
			const handler = jest.fn();
			const wrapper = shallow(
				<ListItem title='ListItem' onPress={handler} />
			);
			expect(wrapper.name()).toMatch(/TouchableHighlight/);
			wrapper.simulate('press');
			expect(handler).toBeCalled();
		});

		it('title props test', () => {
			// 使用children属性时，title属性失效
			const wrapper = shallow(
				<ListItem title='title'>
					<Text>children</Text>
				</ListItem>
			);
			expect(wrapper.contains('title')).toBe(false);
			expect(wrapper.contains('children')).toBe(true);
		});
	});
})
