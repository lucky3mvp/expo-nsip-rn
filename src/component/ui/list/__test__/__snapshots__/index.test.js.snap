// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Tests List test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderBottomWidth": 0.5,
      "borderColor": "#cbd2d9",
      "borderTopWidth": 0.5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#ededed",
        "borderBottomWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#43484d",
            "flex": 1,
            "fontSize": 14,
          }
        }
      >
        ListItem
      </Text>
    </View>
  </View>
</View>
`;

exports[`List Tests ListItem test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderBottomColor": "#ededed",
      "borderBottomWidth": 0.5,
      "flex": 1,
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#43484d",
          "flex": 1,
          "fontSize": 14,
        }
      }
    >
      ListItem
    </Text>
  </View>
</View>
`;
